contracts/Agreement.sol

```diff
@@ -29,11 +29,13 @@ contract Agreement is AccessControl, Ownable {
     uint256   public totalAccounts = 0;
     address[] public optedInAccounts;
     address[] public optedOutAccounts;
+    bool      public contractActive;
 
     mapping (address => bool) public optIn;
     mapping (address => bool) public optOut;
 
     constructor() {
+      contractActive = true;
```

Comment:
if `true` is the default value, it can be initialized in the state variable as opposed to declaring it here:
```
bool public contractActive = true;
```